### PR TITLE
Adds validation of k8s admission plugins

### LINKED ISSUE #6559: Validate Kubernetes Admission Plugins in the Shoot spec
**How to categorize this issue?**

/area usability
/area ops-productivity
/kind enhancement

**What would you like to be added**:
Add validation for the user specified admission plugins in the Shoot spec. Shoot creation or update is denied if the spec contains an invalid plugin config.

**Why is this needed**:
Certain admission plugins are only compatible with certain Kubernetes versions.
eg: `PodSecurityPolicy` is deprecated in kubernetes `v1.25`+
So, if the Shoot owner specifies a plugin in the Shoot spec (`.spec.kubernetes.kubeAPIServer.AdmissionPlugins`) and this plugin is not supported in that particular kubernetes version, this can cause the KubeAPIServer to not start up.

We need something similar to https://github.com/gardener/gardener/issues/3987

### PR DESCRIPTION
**Pull request by plkokanov**
**How to categorize this PR?**

/area usability
/area ops-productivity
/kind enhancement

**What this PR does / why we need it**:
Adds validation of Kubernetes admission plugins. The admission plugins and the versions in which they were added/removed will be maintained by first executing a hack script - `compare-k8s-admission-plugins.sh <old-version> <new-version>` which returns 2 lists of admission plugins. The first one specifies which ones were added in the new version and the second one - which ones were removed in the new version. The lists can then be used to edit  the `admissionPluginsVersionRanges`  map in `pkg/utils/validation/admissionplugins/admissionplugins.go`.
The lists are computed from `https://raw.githubusercontent.com/kubernetes/kubernetes/release-<release>/pkg/kubeapiserver/options/plugins.go` by using only the registered plugins specified in this method: https://github.com/kubernetes/kubernetes/blob/bcea98234f0fdc5ce4e976758977eec72b13bfd4/pkg/kubeapiserver/options/plugins.go#L107 and then getting their names from comments in https://github.com/kubernetes/kubernetes/blob/bcea98234f0fdc5ce4e976758977eec72b13bfd4/pkg/kubeapiserver/options/plugins.go#L64

Basically it works similar to #4149

**Which issue(s) this PR fixes**:
Fixes #6559

**Special notes for your reviewer**:

**Release note**:

```other operator
Kubernetes admission plugins that can be specified in `shoot.kubernetes.apiServer.admissionPlugins` are now validated aginst the kubernetes version of the shoot cluster.
```

### CODE REVIEW

File: hack/compare-k8s-admission-plugins.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+fi
+
+versions=("$1" "$2")
+files=(
+  "pkg/kubeapiserver/options/plugins.go"
+)
+
+# out_dir=$(mktemp -d)
+
+out_dir=/Users/i077286/tmp/out
```
Comment (rfranzke) [inline hack/compare-k8s-admission-plugins.sh]: 👀

File: hack/compare-k8s-admission-plugins.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
```
Comment (rfranzke) [inline hack/compare-k8s-admission-plugins.sh:1]: Adapt https://github.com/gardener/gardener/blob/master/docs/development/new-kubernetes-version.md#adapting-gardener with information about this script.

**Review by rfranzke — COMMENTED** [review]
Looks good generally! Please finish it up.

> However, I had some concerns wrt the handling forbidden and required plugins.

What are your concerns?

**Comment by shafeeqes** [comment]
> What are your concerns?

Plamen wanted to remove the dedicated sets for the required and forbidden plugin checks and just use the fields in this struct:
```golang
// AdmissionPluginVersionRange represents a version range of type [AddedInVersion, RemovedInVersion).
type AdmissionPluginVersionRange struct {
	Forbidden        bool
	Required         bool
	AddedInVersion   string
	RemovedInVersion string
}
```

**Comment by rfranzke** [comment]
Sounds good

**Comment by plkokanov** [comment]
Note that I have not modified this check as it brings a bit more information for the upgrade to v1.25 https://github.com/gardener/gardener/blob/33f339f1709d596ed0e7e5379954a5a856651302/pkg/apis/core/validation/shoot.go#L981-L994
@shafeeqes I could figure out if there is a way to combine it with the new validation if you think that it would be better.

**Comment by rfranzke** [comment]
The failing integration test is because of #6607.

/test pull-gardener-integration

File: hack/local-development/local-garden/run-kube-apiserver [hunk 3]
```diff
@@ -67,4 +67,5 @@ docker run -d --name kube-apiserver -l $LABEL $ADD_HOSTS --network gardener-dev
   --service-account-signing-key-file="/keys/sa.key" \
   --service-account-issuer="local-garden" \
   --service-cluster-ip-range="10.0.0.0/24" \
-  --secure-port=$LISTEN_PORT
+  --secure-port=$LISTEN_PORT \
+  --disable-admission-plugins="ValidatingAdmissionWebhook"
```
Comment (rfranzke) [inline hack/local-development/local-garden/run-kube-apiserver]: Why did you add this?

**Review by rfranzke — COMMENTED** [review]
Nice PR, waiting for feedback of @shafeeqes as well

File: hack/local-development/local-garden/run-kube-apiserver (hunk 3 above)
Comment (plkokanov) [reply hack/local-development/local-garden/run-kube-apiserver]: Added by accident with the last force push :/.

**Review by rfranzke — COMMENTED** [review]
/approve

**Comment by shafeeqes** [comment]
> Nice PR, waiting for feedback of @shafeeqes as well

I will take a look later today.

**Comment by shafeeqes** [comment]
> Note that I have not modified this check as it brings a bit more information for the upgrade to v1.25
> 
> https://github.com/gardener/gardener/blob/33f339f1709d596ed0e7e5379954a5a856651302/pkg/apis/core/validation/shoot.go#L981-L994
> 
> 
> @shafeeqes I could figure out if there is a way to combine it with the new validation if you think that it would be better.

I feel we can leave this here itself.
/lgtm
